zschema/compounds.py

```diff
@@ -192,8 +192,11 @@ def validate(self, value):
             raise DataValidationException("record is not a dict", str(value))
         for subkey, subvalue in value.items():
             if subkey not in self.definition:
-                raise DataValidationException("%s is not a valid subkey of root",
-                                              subkey)
+                port = Port(subkey)
```

Comment:
I fundamentally disagree. This solution is strictly a hack, there's code in place that's intended to handle this situation.